frontend/src/components/Graph.tsx

```diff
@@ -153,17 +161,28 @@ export default class Graph extends React.Component<GraphProps> {
           }
         }
       }
-      displayEdges.push({ from: edgeInfo.v, to: edgeInfo.w, lines });
+      displayEdges.push({
+        color: edge.color,
+        from: edgeInfo.v,
+        isPlaceholder: edge.isPlaceholder,
+        lines,
+        to: edgeInfo.w
+      });
     });
 
     return (
       <div className={css.root}>
         {graph.nodes().map(id => Object.assign(graph.node(id), { id })).map((node, i) => (
-          <div className={classes(css.node, 'graphNode',
+          <div className={classes(node.isPlaceholder ? css.placeholderNode : css.node, 'graphNode',
```

Comment:
How does `node` have `isPlaceholder`? I only see it added to edges elsewhere.

Reply:
`isPlaceholder` is sometimes set within `WorkflowParser` (as of this PR)